lib/unifex/code_generator/utils.ex

```diff
@@ -15,6 +15,22 @@ defmodule Unifex.CodeGenerator.Utils do
     content
   end
 
+  @doc """
+  Replaces special characters such as: `.`, `->` and array brackets e.g. `var_name[i]`
+  with underscores.
+
+  In case of arrays `var_name[i]` results in `var_name_i`.
+  """
+  @spec sanitize_var_name(String.t()) :: String.t()
+  def sanitize_var_name(var_name) do
+    var_name =
+      var_name
+      |> String.replace(".", "_")
+      |> String.replace("->", "_")
+
+    Regex.replace(~r/\[(.*)\]/, var_name, "_\\1")
```

Comment:
1. `String.replace()` accepts a list of patterns
2. It also accepts regexes (https://hexdocs.pm/elixir/String.html#replace/4)
```suggestion
    var_name
    |> String.replace([".", "->"], "_")
    |> String.replace(~r/\[(.*)\]/, "_\\1")
```